panic: runtime error: invalid memory address or nil pointer dereference

### What happened?

The function call `ReadEnvironmentProperty` on a not existing key results in a 

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1045c4fc8]

goroutine 1 [running]:
github.com/pulumi/esc-sdk/sdk/go.(*EscClient).ReadEnvironmentProperty(0x1400013ff08, {0x1046f9b80, 0x140000a9170}, {0x1045d1af6, 0x6}, {0x1045dd612, 0x21}, {0x1045d70f4, 0x15}, {0x1400019c570, ...}, ...)
        /Users/dirien/go/pkg/mod/github.com/pulumi/esc-sdk/sdk@v0.10.5/go/api_esc_extensions.go:149 +0x188
main.main()
        /private/tmp/esc-test/main.go:30 +0x3a0
exit status 2
```

### Example

```go
package main

import (
	"fmt"
	esc "github.com/pulumi/esc-sdk/sdk/go"
	"log"
	"os"
)

func main() {
	accessToken := os.Getenv("PULUMI_ACCESS_TOKEN")
	//projectName := "dirien"
	//orgName := "ediri"
	//envName := "joke-app"
	orgName := "dirien"
	projectName := "esc-secrets-store-csi-driver-demo"
	envName := "csi-secrets-store-app"
	configuration := esc.NewConfiguration()
	escClient := esc.NewClient(configuration)
	authCtx := esc.NewAuthContext(accessToken)

	_, values, err := escClient.OpenAndReadEnvironment(authCtx, orgName, projectName, envName)
	if err != nil {
		log.Fatalf("Failed to open and read environment: %v", err)
	}
	fmt.Println(values)

	env, err := escClient.OpenEnvironment(authCtx, orgName, projectName, envName)

	val, _, err := escClient.ReadEnvironmentProperty(authCtx, orgName, projectName, envName, env.GetId(), "hello")
	if err != nil {
		log.Fatalf("Failed to read environment property: %v", err)
	}
	fmt.Println(val)

}
````


With env:


```env
values:
  app:
    hello: world
    hello-secret:
      fn::secret: xfsdfsd
```

### Output of `pulumi about`

-

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/sdk/go/api_esc_extensions.go
+++ b/sdk/go/api_esc_extensions.go
@@ -146,6 +146,10 @@ func (c *EscClient) OpenAndReadEnvironmentAtVersion(ctx context.Context, org, pr
 // The property is returned along with the resolved value.
 func (c *EscClient) ReadEnvironmentProperty(ctx context.Context, org, projectName, envName, openEnvID, propPath string) (*Value, any, error) {
 	prop, _, err := c.EscAPI.ReadOpenEnvironmentProperty(ctx, org, projectName, envName, openEnvID).Property(propPath).Execute()
+	if prop == nil {
+		return nil, nil, err
+	}
+
 	v := mapValuesPrimitive(prop.Value)
 	return prop, v, err
 }

